nncf/parameters.py

```diff
@@ -0,0 +1,82 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from enum import Enum
+from typing import List, Optional
+
+
+class TargetDevice(Enum):
+    """
+    Describes the target device the specificity of which will be taken
+    into account while compressing in order to obtain the best performance
+    for this type of device.
+
+    :param ANY:
+    :param CPU:
+    :param GPU:
+    :param VPU:
+    """
+
+    ANY = 'ANY'
+    CPU = 'CPU'
+    GPU = 'GPU'
+    VPU = 'VPU'
+
+
+class IgnoredScope:
+    """
+    Dataclass that contains description of the ignored scope.
+
+    The ignored scope specify model sub-graphs which should be excluded from
+    the optimization process such as quantization, pruning and etc.
+
+    For example, you want to exclude some model nodes from the optimization
+    process by node name:
+
+    ```
+    import nncf
+    node_names = ['node_1', 'node_2', 'node_3']
+    ignored_scope = nncf.IgnoredScope(node_names=node_names)
```

Comment:
`node_` looks excessive really. Can we go without it, like `names`, `patterns`, `types`.

Reply:
Thanks for the comment! To be honest, I did as you suggested at first. 

Of course, if we do this, then the attribute names will become shorter, but won't the readability of the code suffer?

IgnoredScope is impersonal by definition. If we change `node_names` -> `names`, the user may have a question: the name of what? Then the user have to go to the documentation... @AlexKoff88, what do you think of the above?

@KodiaqQ @andrey-churkin @vshampor please share your opinion.

Reply:
I think in both cases user should read the documentation about this class before writing some code because the `IgnoredScope` is an advanced feature. The `names`, `patterns`, and `types` look better for me.

Reply:
Based on POT's experience, users understand this field in their own way: one will take this field as it should and `insert the names of the original layers` there, and the other will take it differently and `insert the names of the quantizers (or fake quantize nodes)` there. And both of them may be wrong, because, the names from the NNСFGraph were implied, for example.
My comment is that whatever field name we set, users will perceive it in their own way (and they will be right), so it's better just to write good documentation. We will not be able to come up with a concise and understandable field name for everyone, in this case, it seems to me.

Reply:
Please take a look.

Please do not hesitate if need to do some changes or rollback changes.